arma-rs/src/context.rs

```diff
@@ -1,22 +1,83 @@
-use std::sync::Arc;
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+};
 
 use crossbeam_queue::SegQueue;
 
 use crate::{IntoArma, State, Value};
 
+#[derive(Clone)]
+pub(crate) struct ArmaContext {
+    steam_id: Option<String>,
+    file_source: Option<PathBuf>,
+    mission_name: Option<String>,
+    server_name: Option<String>,
+}
+
+impl ArmaContext {
+    pub(crate) fn new() -> Self {
+        Self {
+            steam_id: None,
+            file_source: None,
+            mission_name: None,
+            server_name: None,
+        }
+    }
+
+    pub(crate) fn with_steam_id(mut self, steam_id: &str) -> Self {
```

Comment:
```rs
enum Caller {
  Steam(u64),
  Singleplayer,
  Dedicated,
}
```

I think 0 will be a dedicated server, but should be confirmed

Reply:
Well that's, fun. What does it return on dedicated?

Reply:
What is the steam id though? Is it the id of the account used to login to steamcmd?

Reply:
That's probably it then, which is a little annoying

Reply:
32bit supports u64 no problem